### PR TITLE
fftw: Add a recipe to build FFTW

### PR DESCRIPTION
**Pull request by jakirkham**
Based off the Archlinux build ( https://projects.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/fftw ).

Basically taking my own recipe for this ( https://github.com/nanshe-org/nanshe-build/tree/master/fftw ) with a few minor tweaks. Mac and Linux. No Windows yet.

### CODE REVIEW

**Comment by jakirkham** [comment]
It seems to complain about tests. However, the `make check` step occurs during the build. How would you like me to address this?

**Comment by jakirkham** [comment]
The build appears to run through. It is merely a matter of picking reasonable tests at this point.

**Comment by pelson** [comment]
> It is merely a matter of picking reasonable tests at this point.

Perhaps you could assert that a .so is available as a result. It's not perfect, but is a start: https://github.com/conda-forge/hdfeos5-feedstock/blob/master/recipe/meta.yaml

**Comment by jakirkham** [comment]
Yeah, I think I like that idea. After all, `make check` is already run during the build. So, just checking to see it is installed seems like enough.

File: recipes/fftw/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package:
+    name: fftw
+    version: "3.3.4"
+
+source:
+    fn: fftw-3.3.4.tar.gz
+    url: http://www.fftw.org/fftw-3.3.4.tar.gz
+    md5: 2edab8c06b24feeb3b82bbb3ebf3e7b3
+
+build:
```
Comment (pelson) [inline recipes/fftw/meta.yaml]: Needs to skip windows.

**Comment by pelson** [comment]
:+1: once the recipe skips windows and has a suitable test (checking an .so exists is reasonable)

File: recipes/fftw/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/fftw/meta.yaml]: Yep, sorry, will remember these in the future.

**Comment by jakirkham** [comment]
Added a test for the expected static libraries as this should work the same on Mac and Linux without getting to gross. Skips Windows. Let me know if you have any more thoughts?

**Comment by ocefpaf** [comment]
I re-started `Travis-CI`, but I am afraid we are exceeding its time limit with this recipe.

**Comment by jakirkham** [comment]
Yeah, it gets pretty close. It ran ok the first time few times I pushed it. Appears it ran ok now.

**Comment by jakirkham** [comment]
Looks like the AppVeyor build was canceled. Though this doesn't actually build anything on Windows so I don't know that it is worth restarting it. Everything else passes.

Is this ok for a merge now?

**Comment by ocefpaf** [comment]
> Looks like the AppVeyor build was canceled. Though this doesn't actually build anything on Windows so I don't know that it is worth restarting it. Everything else passes.

I cancelled it :stuck_out_tongue_winking_eye: 

> Is this ok for a merge now?

:+1:

**Comment by jakirkham** [comment]
Great, thanks @ocefpaf and @pelson. Will do a `pyfftw` recipe after this gets built. Hopefully this will be of some use for others.

**Pull request merged**